base/src/config/configerator.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright Antioch. All rights reserved.
+
+#include <cstdlib>
+#include <fstream>
+
+#include "antioch/base/config/configerator.h"
+#include "antioch/base/constants.h"
+#include <nlohmann/json.hpp>
+
+namespace antioch::base::Configerator {
+
+using json = nlohmann::json;
+using antioch::base::Config;
+
+std::unique_ptr<Config> read_or_exception() {
+  const char* home_dir = getenv("HOME");
+  const std::string path = std::string(home_dir) + CONFIGERATOR_FILE_PATH_FROM_HOME;
+  std::ifstream in(path);
+  json j;
+  in >> j;
+  in.close();
+  if (j.type() != json::value_t::object) {
+    throw ConfigeratorReadException("Wrong json type");
+  }
+  auto config = new Config();
+  j.get_to(*config);
+
+  Config::validateOrThrow(config);
+  return std::unique_ptr<Config>(config);
+}
+
+void write_or_exception(std::unique_ptr<Config> config) {
+  auto c = std::move(config);
```

Comment:
There's gotta be a better way to do these move semantics. Is && not good? What about just const ref since config is this is a synchronous function

Reply:
gonna use const& bc after more consideration this should not be moved